optin/reporting/controllers/completedController.js

```diff
@@ -0,0 +1,218 @@
+const completedDB = require('../models/completed');
+const {
+	hexStringToByte, formatStringSplit, byteToHexString, toRelativeAmount, convertTimestampToDate
+} = require('../utils/ServerUtils');
+const { Parser } = require('json2csv');
+const { NemFacade, SymbolFacade } = require('symbol-sdk').facade;
+
+const isPostOptin = optinTypeFilter => {
+	switch (optinTypeFilter) {
+	case 'pre':
+		return false;
+	case 'post':
+		return true;
+	default:
+		return null;
+	}
+};
+
+const processData = items =>
+	items.map(item => ({
+		optin_id: item.id,
+		isPostoptin: item.is_postoptin,
+		label: item.nemSource.map(props => props.label),
+		nemAddress: item.nemSource.map(props => new NemFacade.Address(hexStringToByte(props.address))
+			.toString()),
+		nemBalance: item.nemSource.map(props => props.balance),
+		nemHeights: item.nemSource.map(props => formatStringSplit(props.height)),
+		nemHashes: item.nemSource.map(props => formatStringSplit(props.hashes)),
+		nemTimestamps: item.nemSource.map(props => formatStringSplit(props.timestamps)),
+		symbolAddress: item.symbolDestination.map(props => new SymbolFacade.Address(hexStringToByte(props.address))
+			.toString()),
+		symbolHeights: item.symbolDestination.map(props => props.height),
+		symbolHashes: item.symbolDestination.map(props => props.hashes),
+		symbolTimestamps: item.symbolDestination.map(props => props.timestamps),
+		symbolBalance: item.symbolDestination.map(props => props.balance)
+	}));
+
+const findMaxInArray = arr => arr.reduce((m, e) => (e > m ? e : m));
+
+const getLatestTimestamps = timestamps => {
+	if (Array.isArray(timestamps))
+		return findMaxInArray(timestamps.map(timestamp => Number(timestamp)));
+	return timestamps ? parseInt(timestamps, 10) : null;
```

Comment:
symbol timestamp should be BitInt

Reply:
In the DB, we have a decimal point for the timestamp, can we use `Math.trunc()` in integer?

Reply:
I will use the `Math.trunc()` it's not affected in CSV.

Reply:
where does decimal point come from?

Reply:
I believe when script import to database.
![image](https://user-images.githubusercontent.com/5649156/168103917-1af70139-03b8-4921-9700-fd6094e13cd3.png)

Reply:
😬 ok ... we probably should have caught that earlier, as those shouldn't be stored as floating point 😬 .